test/acceptance/pages.py

```diff
@@ -215,12 +218,30 @@ def assess(self, options_selected):
         >>> page.assess([0, 2, 1])
 
         """
-        for criterion_num, option_num in enumerate(options_selected):
+        def selector(criterion_num, option_num):
             sel = ".rubric_{criterion_num}_{option_num}".format(
                 criterion_num=criterion_num,
                 option_num=option_num
             )
-            self.q(css=self._bounded_selector(sel)).first.click()
+            return sel
+
+        def select_criterion():
+            for criterion_num, option_num in enumerate(options_selected):
+                sel = selector(criterion_num, option_num)
+                self.q(css=self._bounded_selector(sel)).first.click()
+
+        def criterion_selected():
+            for criterion_num, option_num in enumerate(options_selected):
+                sel = selector(criterion_num, option_num)
+                if not self.q(css=self._bounded_selector(sel))[0].is_selected():
+                    return False
+            return True
+
+        attempts = 0
+        while not criterion_selected() and attempts < 5:
```

Comment:
This retry logic is because sometimes the selection of options in the 2nd training assessment fails (the options do not get selected, and therefore the test times out because the Submit button never enables). I tried waiting on various things to ensure that the second assessment was fully loaded, but nothing helped. Retrying up to at most 5 times seems to have made this test stable on devstack (it very rarely ever failed on Jenkins... the flaky issue was primarily devstack), and it adds very little extra time to the test. However, I fully admit it is still a hack. I just don't have any other ideas at the moment.